Adapt message serialization for Python 3 (#1870)

gleichdick opened the pull request:
### Description

<del>Some changes were missing on #1971, the PR has not been blocked by the CI Pipeline.</del>

Added the missing bits to #1971 which was rolled back.

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

v4hn commented:
Sorry, I cannot verify this cleanup commit right now (CI does not trigger and locally I get linking errors that might not be related to either of these patches). Instead I reverted your broken cherry-pick for now.

Could I ask you to reapply the full and clean cherry-pick in this pull-request?

gleichdick commented:
Okay, I rebased and squashed onto the reverted melodic-devel branch.
`WIP` is added to prevent a merge before the CI is fixed, the patch itself is ready as the [CI shows](https://travis-ci.org/github/ros-planning/moveit/builds/666505117).
Sorry for messing up the branch
